openapi-communication-service.yaml

```diff
@@ -0,0 +1,623 @@
+      headers:
+        Content-Type:
+          schema:
+            example: application/problem+json
+            type: string
+          description: Indicates the original media type of the resource
+    RequestTooLarge:
+      description: The payload for the body of the request is too large (RequestEntityTooLargeException)
+      content:
+        application/problem+json:
+          schema:
+            $ref: ./models/Problems.yaml
+          examples:
+            Example 1:
+              value:
+                errors:
+                  - code: 42R
+                    title: Request Entity Too Large
+                    status: 413
+                    detail: The payload for the body of the request is too large
+                    instance: /communications/6090146f-1023-4f7c-8eed-10a045d37ae8
+      headers:
+        Content-Type:
+          schema:
+            example: application/problem+json
+            type: string
+          description: Indicates the original media type of the resource
+    TooManyRequests:
+      description: Too Many Requests
+      content:
+        application/problem+json:
+          schema:
+            $ref: ./models/Problems.yaml
+          examples:
+            Example 1:
+              value:
+                errors:
+                  - code: 13L
+                    title: Too many requests
+                    status: 429
+                    detail: You reached your request limits
+                    instance: /communications/6090146f-1023-4f7c-8eed-10a045d37ae8
+      headers:
+        Content-Type:
+          schema:
+            example: application/problem+json
+            type: string
+          description: Indicates the original media type of the resource
+  schemas:
+    DocumentCreatedResponse:
+      title: DocumentCreatedResponse
+      x-stoplight:
+        id: 1m6vp0nti15g2
+      type: object
+      required:
+        - id
+      properties:
+        id:
+          type: string
+          x-stoplight:
+            id: 1un77btvbbjmv
+          format: uuid
+          x-faker: datatype.uuid
+          example: e2b5d3d9-8e08-4459-b6c5-dc7a5118a3d9
+          description: The uuid of the communications created.
+    CommonCommunicationBody:
+      title: ''
+      x-stoplight:
+        id: w6wjeregmzh7i
+      type: object
+      required:
+        - communication_definition_name
+        - document_type
+        - communication_type
+      properties:
+        communication_definition_name:
+          type: string
+          x-stoplight:
+            id: njrjlpje4adso
+          description: The unique name of the communication definition object created in RapidCX
+          example: business letter 0001
+        document_type:
+          type: string
+          x-stoplight:
+            id: bojpimqqrrltn
+          example: letter
+          description: 'The category of the document being created (e.g., letters, statements, claims)'
+        communication_type:
```

Comment:
As mentioned above, I had to change the `communication_type` because java generator protested ...